packages/mirrorful/editor/src/pages/_app.tsx

```diff
@@ -18,7 +24,49 @@ if (typeof window !== 'undefined') {
 }
 
 export default function MyApp({ Component, pageProps }: AppProps) {
+  const [isLoading, setIsLoading] = useState(true)
   const router = useRouter()
+  const {
+    setColors,
+    setTypography,
+    setShadows,
+    setFileTypes,
+    setShowOnBoarding,
+    shouldForceSkipOnboarding,
+    showOnBoarding,
+  } = useMirrorfulStore((state: MirrorfulState) => state)
+  // to fetch data
+
+  const timeout = useRef<NodeJS.Timeout | null>(null)
+  const fetchStoredData = useCallback(async () => {
+    try {
+      const data = await fetchStoreData()
+
+      if (
+        !Object.keys(data).length ||
+        !data.tokens.colorData ||
+        data.tokens.colorData.length === 0
+      ) {
+        setShowOnBoarding(true)
+        return
+      }
+
+      setColors(data.tokens.colorData ?? [])
+      setTypography(data.tokens.typography)
+      setShadows(data.tokens.shadows ?? defaultShadows)
+      setFileTypes(data.files)
+    } catch (e) {
+      // TODO: Handle error
+    } finally {
+      timeout.current = setTimeout(() => {
```

Comment:
What is this timeout logic for?

Reply:
This timeout is used for the splash screen. So all data can be loaded in before using the app.